pandas/tests/io/test_sql.py

```diff
@@ -503,10 +507,14 @@ class _TestSQLApi(PandasSQLTest):
     flavor = 'sqlite'
     mode = None
 
-    @pytest.fixture(autouse=True)
-    def setup_method(self, datapath):
+    def setup_connect(self):
```

Comment:
Any reason this is a dedicated method? Seems like a simple enough expression to have just included in the setup method alone

Reply:
`load_iris_data` fixture basically requires that self has a conn attribute set, see line 262. I basically wanted to separate out the connection setup for each test classes. Since `_TestSQLAlchemy` already did this I though it would be nice to keep it consistent